libass/ass.c

```diff
@@ -1522,17 +1522,28 @@ ASS_Track *ass_new_track(ASS_Library *library)
 
 int ass_track_set_feature(ASS_Track *track, ASS_Feature feature, int enable)
 {
-    switch (feature) {
-    case ASS_FEATURE_INCOMPATIBLE_EXTENSIONS:
-        //-fallthrough
+    bool *features[] = {
+        [ASS_FEATURE_INCOMPATIBLE_EXTENSIONS] = NULL,
 #ifdef USE_FRIBIDI_EX_API
-    case ASS_FEATURE_BIDI_BRACKETS:
-        track->parser_priv->bidi_brackets = !!enable;
+        [ASS_FEATURE_BIDI_BRACKETS] = &track->parser_priv->bidi_brackets,
 #endif
+        [ASS_FEATURE_ALWAYS_WRAP] = &track->always_wrap,
+    };
+
+    if (feature == ASS_FEATURE_INCOMPATIBLE_EXTENSIONS) {
```

Comment:
Oh wait, this won't work if `!USE_FRIBIDI_EX_API`, because it still takes up a space in the enum but isn't used. Hmm, let me think about a better way.